paper-api/src/main/java/org/bukkit/inventory/PredicateChoiceImpl.java

```diff
@@ -0,0 +1,38 @@
+package org.bukkit.inventory;
+
+import com.google.common.base.Preconditions;
+import org.jspecify.annotations.NullMarked;
+import java.util.function.Predicate;
+
+@NullMarked
+record PredicateChoiceImpl(Predicate<ItemStack> stackPredicate, ItemStack exampleStack) implements RecipeChoice.PredicateChoice {
+
+    public PredicateChoiceImpl(Predicate<ItemStack> stackPredicate, ItemStack exampleStack) {
+        Preconditions.checkArgument(stackPredicate != null, "The item predicate cannot be null");
+        Preconditions.checkArgument(exampleStack != null, "The example stack cannot be null");
+        Preconditions.checkArgument(!exampleStack.isEmpty(), "Cannot have empty/air example stack");
+
+        this.stackPredicate = stackPredicate;
+        this.exampleStack = exampleStack.clone();
+    }
```

Comment:
```suggestion
    public PredicateChoiceImpl {
        Preconditions.checkArgument(stackPredicate != null, "The item predicate cannot be null");
        Preconditions.checkArgument(exampleStack != null, "The example stack cannot be null");
        Preconditions.checkArgument(!exampleStack.isEmpty(), "Cannot have empty/air example stack");
        exampleStack = exampleStack.clone()
    }
```
can use a fancy record compact constructor here since its just preconditions